### PR TITLE
DOC: Conform to numpydoc + uniformity.

### PR DESCRIPTION
**Pull request by Carreau**
This update a couple of docstring to math numpydoc syntax.
For parameter and return types, space before colon is important for
numpydoc to properly find the parameter name and type.

Try to consistently indent 4 spaces in RST.

Try to be consistent and spacing between sections and parameters.

### CODE REVIEW

File: scipy/cluster/hierarchy.py [hunk 1]
```diff
@@ -3491,17 +3490,16 @@ def _dendrogram_calculate_info(Z, p, truncate_mode,
...
+        * w is the amount of space used for the subtree (in independent
+          variable units)
 
-      * h is the height of the subtree in dependent variable units
+        * h is the height of the subtree in dependent variable units
 
-      * md is the ``max(Z[*,2]``) for all nodes ``*`` below and including
-        the target node.
+        * md is the ``max(Z[*,2]``) for all nodes ``*`` below and including
+          the target node.
```
Comment (tirthasheshpatel) [inline scipy/cluster/hierarchy.py:3502]: Is this the right/standard way to create bullet lists in `numpydoc`? I always thought it was:

```
Description of something

* first point
* second point
* third point

```

What I see here is:

```
Description of something
  * first point

  * second point

  * third point

```

This probably doesn't matter much here as it isn't rendered on the website.

Comment (Carreau) [reply scipy/cluster/hierarchy.py:3502]: Probably not, right now I'm only making sure that Numpydoc does not crash when parsing it, and I basically do : 

 Parse -> Fix -> reformat -> ReParse , and  make sure the -re-parsing give the same data sctructure than after the "fix" step.

Numpydoc "sees" the following for the return section regardless of the whitespaces.

```
In [6]: ds.NumpyDocString(s)._parsed_data
Out[6]:
{'Signature': '',
...
 'Parameters': [],
 'Returns': [Parameter(name='', type='A tuple (left, w, h, md), where:', desc=['* left is the independent variable coordinate of the center of the', '  the U of the subtree', '', '* w is the amount of space used for the subtree (in independent', '  variable units)', '', '* h is the height of the subtree in dependent variable units', '', '* md is the ``max(Z[*,2]``) for all nodes ``*`` below and including', '  the target node.'])],
 'Yields': [],
 ...
```

Comment (tirthasheshpatel) [reply scipy/cluster/hierarchy.py:3502]: > Parse -> Fix -> reformat -> ReParse , and make sure the -re-parsing give the same data sctructure than after the "fix" step.

Oh, OK. That makes sense.

> Numpydoc "sees" the following for the return section regardless of the whitespaces.

Hmm, didn't know `numpydoc` doesn't care about whitespaces. Thanks for the clarification!

**Review by rgommers — APPROVED** [review]
Thanks @Carreau!

**Pull request merged**